common/scala/src/main/scala/whisk/core/database/memory/MemoryArtifactStore.scala

```diff
@@ -131,14 +131,13 @@ class MemoryArtifactStore[DocumentAbstraction <: DocumentSerializer](dbName: Str
 
     val f = Future.fromTry(t)
 
-    f.onFailure({
+    f.failed.foreach {
       case _: DocumentConflictException =>
         transid.finished(this, start, s"[PUT] '$dbName', document: '$docinfoStr'; conflict.")
-    })
+      case _ =>
+    }
 
-    f.onSuccess({
-      case _ => transid.finished(this, start, s"[PUT] '$dbName' completed document: '$docinfoStr'")
-    })
+    f.foreach(_ => transid.finished(this, start, s"[PUT] '$dbName' completed document: '$docinfoStr'"))
```

Comment:
Can we use `onComplete` everywhere there's the success/failure case?

Reply:
Done